extensions/s3/src/main/java/io/deephaven/extensions/s3/S3CompletableOutputStream.java

```diff
@@ -503,17 +507,20 @@ private void abortMultipartUpload() throws IOException {
         }
 
         // Initiate the abort request
-        final AbortMultipartUploadRequest abortRequest = AbortMultipartUploadRequest.builder()
+        final AbortMultipartUploadRequest.Builder builder = AbortMultipartUploadRequest.builder()
                 .bucket(uri.bucket().orElseThrow())
                 .key(uri.key().orElseThrow())
-                .uploadId(uploadId)
-                .build();
+                .uploadId(uploadId);
+        final Duration writeTimeout = s3Instructions.writeTimeout();
+        builder.overrideConfiguration(b -> addTimeout(b, writeTimeout));
+        final AbortMultipartUploadRequest abortRequest = builder.build();
```

Comment:
We should be able to extract this common logic, some of the same patterns we see used in `io.deephaven.extensions.s3.S3ClientFactory` (ie, around the generics and the builder types)